lib/vrl/stdlib/src/is_empty.rs

```diff
@@ -0,0 +1,139 @@
+use vrl::prelude::*;
+
+fn is_empty(value: Value) -> Resolved {
+    let empty = match value {
+        Value::Object(v) => v.is_empty(),
+        Value::Array(v) => v.is_empty(),
+        Value::Bytes(v) => v.is_empty(),
+        value => {
+            return Err(value::Error::Expected {
+                got: value.kind(),
+                expected: Kind::array(Collection::any())
+                    | Kind::object(Collection::any())
+                    | Kind::bytes(),
+            }
+            .into())
+        }
+    };
+
+    Ok(empty.into())
+}
+
+#[derive(Clone, Copy, Debug)]
+pub struct IsEmpty;
+
+impl Function for IsEmpty {
+    fn identifier(&self) -> &'static str {
+        "is_empty"
+    }
+
+    fn parameters(&self) -> &'static [Parameter] {
+        &[Parameter {
+            keyword: "value",
+            kind: kind::OBJECT | kind::ARRAY | kind::BYTES,
+            required: true,
+        }]
+    }
+
+    fn examples(&self) -> &'static [Example] {
+        &[
+            Example {
+                title: "empty string",
+                source: r#"is_empty("")"#,
+                result: Ok("true"),
+            },
+            Example {
+                title: "empty array",
+                source: r#"is_empty([])"#,
+                result: Ok("true"),
+            },
+            Example {
+                title: "empty object",
+                source: r#"is_empty({})"#,
+                result: Ok("true"),
+            },
+            Example {
+                title: "non-empty array",
+                source: r#"is_empty([null])"#,
```

Comment:
This feels like a weird example to use with `null` - (but this doesn't get shown to users, just tested?)

Reply:
Yeah, that is weird to use `null` there. It doesn't get shown to users, but I'll update it anyway. This also makes me realize I forgot to update the doc.

Reply:
Updated in [d55680c](https://github.com/vectordotdev/vector/pull/9732/commits/d55680c0e2ef0db35758bfbc334f2f283e8e3a1d)